### ISSUE TITLE
Discarded assignments has false positives

### ISSUE THREAD

**Issue by GeorgiosDelkos** [camden-smallwood/sway-analyzer#8]
Discarded assignments has false positives.

```rust
let mut is_valid_key = false;
let mut i = 0;
while i < 9 {
    if key == module_keys[i] {
        is_valid_key = true;
        break;
    }
    i += 1;
}
```
